homeassistant/components/shelly/config_flow.py

```diff
@@ -262,6 +264,53 @@ async def async_step_confirm_discovery(
             errors=errors,
         )
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+        assert self.entry is not None
+        host = self.entry.data[CONF_HOST]
+
+        if user_input is not None:
+            info = await self._async_get_info(host)
+            if self.entry.data.get("gen", 1) == 2:
```

Comment:
Should this not match line 300? Or you might get unexpected behavior when gen is neither 1 nor 2.
```suggestion
            if self.entry.data.get("gen", 1) != 1:
```

Reply:
This is correct.
Gen2 devices username is hardcoded to "admin.
For older Gen1 devices we did not store the Gen so it might return none.
We can change however here or in line 300 so that the `if` path will be the same (reverse the if else and compare to 2), but I don't worth the change.

Reply:
You're right `if self.entry.data.get("gen", 1) != 1:` will be cleaner and more understandable.